cmake: xcc: unset GCC_NO_G_FLAG	after use

dcpleung opened the pull request:
GCC_NO_G_FLAG is used to skip "-g" to XCC Clang	as the older
version	of Clang has a bug with	"-g". This unsets the flag
after inclusion	of clang/compiler_flags.cmake to limit
the lifetime of the flags here.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

tejlmand commented on cmake/compiler/xcc/compiler_flags.cmake:
```diff
@@ -6,6 +6,10 @@ if(CC STREQUAL "clang")
 
   include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)
+
+  if($ENV{XCC_NO_G_FLAG})
+    unset(GCC_NO_G_FLAG 1)
+  endif()
```
Sorry, this was not what I meant with this comment: https://github.com/zephyrproject-rtos/zephyr/pull/42017#discussion_r795544812

what I meant was that you should do this instead:
```
  include(${ZEPHYR_BASE}/cmake/compiler/clang/compiler_flags.cmake)

  # Now, let's overwrite the flags that are different in xcc/clang.
  if($ENV{XCC_NO_G_FLAG})
    set_compiler_property(PROPERTY debug)
  endif()
```

and then revert this commit:
https://github.com/zephyrproject-rtos/zephyr/commit/deeb98da5328c6fbdfc4ad1c7638a205d86f83e7

dcpleung replied:
Updated, and reverted the old commit.

dcpleung committed:
Revert "cmake: compiler/xcc: omit -g if needed for Clang"

This reverts commit deeb98da5328c6fbdfc4ad1c7638a205d86f83e7.

A less invasive change has been implemented which does not
require changes to the GCC compiler flag file. So revert
the commit.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

dcpleung committed:
cmake: compiler/xcc: override and clear debug property for clang

Some older versions of XCC Clang would result in the following
error during compilation:

  /tmp/file.s: Assembler messages:
  /tmp/file.s:20: Error: file number 1 already allocated
  clang-3.9: error: Xtensa-as command failed with exit code 1

due to a bug in LLVM: https://bugs.llvm.org/show_bug.cgi?id=11740.
This is fixed in upstream, https://reviews.llvm.org/D20002, in 2016.
However, it seems that it is only fixed after XCC RI-2018.0.
Instead of blanket disabling usage of '-g', use an environment
variable "XCC_NO_G_FLAG" to disable usage of flag '-g' to workaround
this issue. This needs to be manually set because there is no way to
know which XCC version is being used, and compiler flag checking for
'-g' would not result in error (and thus '-g' is not ignored).
This is only needed for older XCC Clang. For sufficiently new XCC
verisons, there is no need for this.

Note that this is an alternative implementation to commit
deeb98da5328c6fbdfc4ad1c7638a205d86f83e7. This one does not
alter the GCC flag file, and instead, simply clear the debug
compiler property so "-g" is not used.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

tejlmand reviewed (approved):
lgtm
